hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -298,6 +247,26 @@ public StorageStatistics provide() {
         LOG.debug("Using metadata store {}, authoritative={}",
             getMetadataStore(), allowAuthoritative);
       }
+
+      ExecutorService downloadExecutorService = Executors.newFixedThreadPool(
+              intOption(conf, MULTIPART_DOWNLOAD_NUM_THREADS, DEFAULT_MULTIPART_DOWNLOAD_NUM_THREADS, 0),
+              new ThreadFactoryBuilder().setNameFormat("multipart-download-%d").build());
+      multipartDownloader = new MultipartDownloader(
+              longOption(conf, MULTIPART_DOWNLOAD_PART_SIZE, DEFAULT_MULTIPART_DOWNLOAD_PART_SIZE, 0),
+              MoreExecutors.listeningDecorator(downloadExecutorService),
+              new PartDownloader() {
+                @Override
+                public InputStream downloadPart(String bucket, String key, long rangeStart, long rangeEnd) {
+                  String serverSideEncryptionKey = getServerSideEncryptionKey(getConf());
+                  GetObjectRequest request = new GetObjectRequest(bucket, key);
+                  if (S3AEncryptionMethods.SSE_C.equals(serverSideEncryptionAlgorithm) && StringUtils.isNotBlank(serverSideEncryptionKey)) {
+                    request.setSSECustomerKey(new SSECustomerKey(serverSideEncryptionKey));
+                  }
+                  return s3.getObject(request.withRange(rangeStart, rangeEnd - 1)).getObjectContent();
```

Comment:
expect to move to `invoker.retry` around all s3 calls, with some  retry policy annotations on the method to enable an audit of the policy (no retries over retries, essentially)